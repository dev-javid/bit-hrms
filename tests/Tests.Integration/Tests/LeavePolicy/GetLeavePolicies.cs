namespace Tests.Integration.Tests.LeavePolicy
{
    public class GetLeavePolicies(TestWebApplicationFactory<Program> factory) : IntegrationTest(factory)
    {
        private const string Route = "/api/leave-policy?page=1&limit=10";

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async Task Should_Return_Data()
        {
            await LoginAsCompanyAdminAsync();
            await FeedDataAsync("Tests/LeavePolicy/GetLeavePolicies.sql");
            using HttpResponseMessage response = await Client.GetAsync(Route);

            response.StatusCode.Should().Be(HttpStatusCode.OK);
            var result = await response.Content.ReadAsStringAsync();
            Approvals.VerifyJson(result);
        }
    }
}
