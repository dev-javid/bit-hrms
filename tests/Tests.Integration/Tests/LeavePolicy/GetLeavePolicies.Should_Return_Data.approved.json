{
    "leavePolicyId": 1,
    "holidays": 2,
    "casualLeaves": 3,
    "earnedLeavesPerMonth": 1
}